examples/kalshi-polymarket-vwap/src/execution_phase.rs

```diff
@@ -0,0 +1,143 @@
+ * then calculates a volume-weighted average price between the two platforms.
+ * Currently works with binary prediction markets and focuses on the "yes" outcome price.
+ * Returns the volume-weighted average price as the final result.
+ */
+pub fn execution_phase() -> Result<()> {
+    // Retrieve the input parameters for the data request (DR).
+    // Expected to be a market identifier that works for both Kalshi and PolyMarket APIs.
+    let dr_inputs_raw = String::from_utf8(Process::get_inputs())?;
+    let dr_inputs_trimmed = dr_inputs_raw.trim();
+
+    let market_tickers: Vec<&str> = dr_inputs_trimmed.split(',').collect();
+
+    log!(
+        "Fetching market data from Kalshi and PolyMarket for market: {} and {}",
+        market_tickers[0],
+        market_tickers[1]
+    );
+
+    // Step 1: Fetch Kalshi market data (yes bid price and volume)
+    let kalshi_market_response = http_fetch(
+        format!(
+            "https://api.elections.kalshi.com/trade-api/v2/markets/{}",
+            market_tickers[0]
+        ),
+        None,
+    );
+
+    // Check if the market request was successful
+    if !kalshi_market_response.is_ok() {
+        elog!(
+            "market HTTP Response was rejected: {} - {}",
+            kalshi_market_response.status,
+            String::from_utf8(kalshi_market_response.bytes)?
+        );
+        Process::error("Error while fetching market information".as_bytes());
+        return Ok(());
+    }
+
+    // Parse market informationmarket_response
+    let kalshi_market_data =
+        serde_json::from_slice::<KalshiMarketResponse>(&kalshi_market_response.bytes)?;
+    log!(
+        "Fetched Kalshi Price (YES BID): {} cents with volume {}",
+        kalshi_market_data.market.yes_bid_dollars,
+        kalshi_market_data.market.volume
+    );
+
+    let kalshi_yes_bid_dollars = kalshi_market_data.market.yes_bid_dollars.parse::<f64>()?;
+
+    // Step 2: Fetch PolyMarket market data (yes outcome price and volume)
+    let polymarket_market_response = http_fetch(
+        format!(
+            "https://gamma-api.polymarket.com/markets/{}",
+            market_tickers[1]
+        ),
+        None,
+    );
+
+    // Check if the market request was successful
+    if !polymarket_market_response.is_ok() {
+        elog!(
+            "market HTTP Response was rejected: {} - {}",
+            polymarket_market_response.status,
+            String::from_utf8(polymarket_market_response.bytes)?
+        );
+        Process::error("Error while fetching market information".as_bytes());
+        return Ok(());
+    }
+
+    // Parse market informationmarket_response
+    let poly_market_market_data =
+        serde_json::from_slice::<PolyMarketMarket>(&polymarket_market_response.bytes)?;
+    let outcome_prices_array: Vec<String> =
+        serde_json::from_str(&poly_market_market_data.outcome_prices)?;
+    let polymarket_yes_price = outcome_prices_array[0].parse::<f64>()?; // 0 = yes price
```

Comment:
Similarly, we should turn on `preserve_order` feature since we parse an array, and otherwise there's no guarantees on order.